### DISCUSSION TITLE
updating versions in Linux

### DISCUSSION THREAD

**Discussion by relogi** [vkbo/novelWriter#719] [Help] [answered]
I just downloaded 1.1.3, 
(threw out the 1.1.1 folder)
 cd 'd into the 1.1.3 folder and 
ran setup.py install 
(and setup.py xdg-install); 
everything looked good on the command line. 
Opened up novelWriter (clicking on the icon) and I am now being told I'm running 1.0.3, a version i thought I'd disposed of when I installed 1.1.1. Any thoughts on how zombie novelWriter is haunting me?

I hate wasting people's time on what is most likely an error between keyboard and chair, but thought it worth bringing up in the context of updating the program on linux...

relogi

**Comment by vkbo** [comment, marked as answer]
I guess the simplest solution is to look what the path in the `novelwriter.desktop` file points to. The file should be in either `~/.local/share/applications` or in `/usr/share/applications/`. It is not clear from your description where you're clicking the icon from. It may also be a copy of an older version of the `.desktop` file.

Basically, running these two commands from command line:
```
cat ~/.local/share/applications/novelwriter.desktop
cat /usr/share/applications/novelwriter.desktop
```

Should produce at least one result. The `Exec=` line should reveal where it is pointing.

The `setup.py install` command copies novelWriter to the system's Python package location. This is not really intended for standalone applications in the first place so I don't do this myself. Personally, I'd recommend not running the install step at all. I updated the documentation a while back to reflect this. I keep my working copy of novelWriter, and other applications that don't come from the OS repo, in a software folder. Then you can run `./setup.py xdg-install` from the folder you want to be the active version. If you don't run the step with the `sudo` command, it should update the same `~/.local/share/applications/novelwriter.desktop` file each time.

**Reply by relogi** [reply]
Thank you... Got it. I'm actually diving into the whole Python package location thing now that you've shown me how inadequate my understanding is... has been... It shows you how little thought I was putting into the set up process that I was/have been doing both the setup.py install AND the setup.py xdg-install inside the downloaded version of novelWriter... pip uninstall was how i managed to dig myself out of ignorance this time... your patience with me is amazing,..
relogi

**Reply by vkbo** [reply]
Hah, well, there's nothing wrong with running both the `install` step and `xdg-install`. It's just that the latter is enough. But glad I could help!
